### PR TITLE
Created component CodePreview

### PR DESCRIPTION
**Pull request by rodrigocorsi2**
Created component CodePreview using highlight.js.
Used this component to view inline code and view code of attachments.

Previous PR: [#2335](https://github.com/mattermost/platform/pull/2335)

img1

img2

### CODE REVIEW

**Comment by rodrigocorsi2** [comment]
Please test:
1. Attachments with small and large code
2. Test inline code
3. Test html injection

**Comment by lfbrock** [comment]
Thanks @rodrigocorsi2, @hmhealey can you help with testing/reviewing?

File: web/react/components/code_preview.jsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                    formatMessage={this.props.formatMessage}
+                />
+            );
+        }
+
+        return <div dangerouslySetInnerHTML={{__html: CodePreview.formatCode(this.state.lang, this.state.code, this.props.filename)}}/>;
+    }
+}
+
+function getlang(filename) {
```
Comment (hmhealey) [inline web/react/components/code_preview.jsx]: Can you rename this to getLang so we can keep it consistent with everything else being in camelCase?

File: web/react/components/code_preview.jsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+    handleReceivedCode(data) {
+        this.setState({code: data});
+    }
+
+    handleReceivedError() {
+        this.setState({canLoad: false});
+    }
+
+    static formatCode(lang, data, filename) {
```
Comment (hmhealey) [inline web/react/components/code_preview.jsx]: Can you move these static functions, the getlang function, and all the highlighting imports into their own file in the web/react/utils folder? Something like `web/react/utils/syntaxHightlighting.jsx`? It feels weird to have code in utils calling out to component code. That'll also hide the endless highlight.js imports.

**Comment by hmhealey** [comment]
Looks good other than some organizational things. I'm also spinning up my own spinmint instance since mattermod seems to be having issues with this PR

**Comment by rodrigocorsi2** [comment]
Hi @hmhealey, no problem I'll do these changes.
Will I use `syntaxHightlighting.jsx` or `syntax_hightlighting.jsx` like other files?

**Comment by hmhealey** [comment]
Oh right that's how we name those files :)

And great job. It looks awesome!

**Comment by crspeller** [comment]
@rodrigocorsi2 Looks really good. But needs a rebase.

**Comment by rodrigocorsi2** [comment]
Hi @crspeller! rebase is done

**Comment by rodrigocorsi2** [comment]
Squash commits

**Comment by crspeller** [comment]
+1 But it appears that it needs rebase.

**Comment by rodrigocorsi2** [comment]
Hi @crspeller! rebase is done

**Pull request merged**